skopt/plots.py

```diff
@@ -410,5 +484,227 @@ def plot_evaluations(result, bins=20, dimensions=None):
+    -------
+    * `fig`: [`matplotlib.figure.Figure`]:
+        The Matplotlib Figure-object.
+        For example, you can save the plot by calling `fig.savefig('file.png')` 
+    """
+
+    # Get the search-space instance from the optimization results.
+    space = result.space
+
+    # Get the dimension-object, its index in the search-space, and its name.
+    dimension1, index1, dimension_name1 = space.get_dimension(id=dimension_id1)
+    dimension2, index2, dimension_name2 = space.get_dimension(id=dimension_id2)
+
+    # Ensure dimensions are not Categorical.
+    if type(dimension1) == Categorical or type(dimension2) == Categorical:
+        raise ValueError("Categorical dimension is not supported.")
+
+    # Get the samples from the optimization-log for the relevant dimensions.
+    samples1 = _get_samples_dimension(result=result, index=index1)
+    samples2 = _get_samples_dimension(result=result, index=index2)
+
+    # Get the best-found samples for the relevant dimensions.
+    best_sample1 = result.x[index1]
+    best_sample2 = result.x[index2]
+
+    # Get the last fitted model for the search-space.
+    last_model = result.models[-1]
+
+    # Get new random samples from the search-space and transform if necessary.
+    sample_points = space.rvs(n_samples=n_samples)
+    sample_points = space.transform(sample_points)
+
+    # Estimate the objective function for these sampled points
+    # using the last fitted model for the search-space.
+    xi, yi, zi = partial_dependence(space=space,
+                                    model=last_model,
+                                    i=index2, j=index1,
+                                    sample_points=sample_points,
+                                    n_points=n_points)
+
+    # Start a new plot.
+    fig = plt.figure()
+
+    # Scale for the z-axis of the contour-plot. Either Log or Linear (None).
+    locator = LogLocator() if zscale == 'log' else None
+
+    # Use the min and max values of the objective function from the
+    # optimization results to normalize the color-gradient across plots for
+    # different choices of dimensions.
+    # TODO: I'm not sure if this should be used?
+    vmin = np.min(result.func_vals)
+    vmax = np.max(result.func_vals)
+
+    # Plot the contour-landscape for the objective function.
+    plt.contourf(xi, yi, zi, levels, locator=locator, cmap='viridis_r',
+                 vmin=vmin, vmax=vmax)
+
+    # Plot all the parameters that were sampled during optimization.
+    # These are plotted as small black dots.
+    plt.scatter(samples1, samples2, c='black', s=10, linewidths=1)
+
+    # Plot the best parameters that were sampled during optimization.
+    # These are plotted as a big red star.
+    plt.scatter(best_sample1, best_sample2,
+                c='red', s=50, linewidths=1, marker='*')
+
+    # Use the dimension-names as the labels for the plot-axes.
+    plt.xlabel(dimension_name1)
+    plt.ylabel(dimension_name2)
+
+    return fig
```

Comment:
most of the pandas plotting functions return an axis. What other examples of libraries are there that return plots? I think we should try to be consistent with what other libraries do. I'd at least return `fig, axis`

Reply:
As I said above, I never really understood Matplotlib, but when I ran your plotting-functions in a python-file I couldn't figure out how to e.g. save the plot without having the `fig`-object. When using the functions in a Jupyter Notebook they can work either way with returning `fig` or `axis`. I also thought about returning both  `fig, axis` but I didn't think it was necessary. I can give it a try and see if it works in Jupyter.